src/agoradatatools/etl/transform/immunohisto_transform.py

```diff
@@ -212,6 +216,72 @@ def _extract_age_num(age_str: str) -> float:
         raise ValueError(f"Invalid age value: {age_str!r}") from exc
 
 
+def _create_measure_order_key(
+    datasets: Dict[str, pd.DataFrame],
+    dataset_name: str,
+    config_key: str = "immunohisto_measure_order",
+) -> callable:
+    """
+    Create a sorting key function for evidence_type based on the measure order configuration.
+
+    This function loads the measure order configuration from datasets and creates a sorting
+    key function. Evidence types listed in the config will be sorted in that order, while
+    unlisted types will be sorted alphabetically after the listed types.
+
+    Args:
+        datasets: Dictionary of dataset names mapped to their DataFrames
+        dataset_name: Name of the dataset ('biomarkers' or 'pathology')
+        config_key: Key to look for in datasets dictionary (defaults to 'immunohisto_measure_order')
+
+    Returns:
+        A function that takes an evidence_type string and returns a tuple that
+        can be used for sorting. The tuple is (order_index, evidence_type) where
+        order_index is the position in the measure order list (or a large number for
+        unlisted types to sort them after listed ones).
+
+    Raises:
+        ValueError: If the config is not found in datasets or has invalid structure
+
+    Example:
+        >>> datasets = {"immunohisto_measure_order": config_df}
+        >>> key_func = _create_measure_order_key(datasets, "biomarkers")
+        >>> sorted(["Soluble A&beta;40", "Other Type", "NfL"], key=key_func)
+        ["NfL", "Soluble A&beta;40", "Other Type"]
+    """
+    # Load and validate config from datasets
+    if config_key not in datasets:
+        raise ValueError(
+            "Measure order configuration 'immunohisto_measure_order' is required but not found in datasets"
+        )
+
+    config_df = datasets[config_key]
+
+    # Validate that required columns are present
+    if not all(col in config_df.columns for col in ["dataset_name", "evidence_type"]):
```

Comment:
Shouldn't this have been validated already with `check_required_datasets_and_columns` in the main transform function?